js_sdk/end_idle_engagements.js

```diff
@@ -0,0 +1,32 @@
+// Reference:
+// https://js-sdk-docs.salemove.com/class/Engagement.html
+// Running the code snippet below before starting an engagement
+// allows to track Visitor activity to end the ongoing engagement
+// when Visitor has been idle (no mouse or keyboard event recorded)
+// for some given amount of time
+
+sm.getApi({version: 'v1'}).then(function(salemove) {
+  var onEngagementStarted = function(engagement) {
+    var idleTime = 0;
+    var timerIncrement = function() {
+      idleTime = idleTime + 1;
+      if (idleTime >= 29) {
+        engagement.end();
+      }
+    }
+    var idleInterval = setInterval(timerIncrement, 1000);
+    $(window).mousemove(function(e) {
+      idleTime = 0;
+    });
+    $(window).keypress(function(e) {
+      idleTime = 0;
+    });
+    $(window).scroll(function(e) {
+      idleTime = 0;
+    });
+    salemove.addEventListener(salemove.EVENTS.ENGAGEMENT_END, function() {
```

Comment:
This should listen for the `END` event on the `engagement` to work correctly with multiengage and other cases.

Reply:
Not sure if I understood correctly but replacing 'salemove.EVENTS.ENGAGEMENT_END' with 'engagement.EVENTS.END' throws the following error in the browser console:

got Uncaught (in promise) Error {cause: "internal error"}

Reply:
Did you use `engagement.addEventListener(engagement.EVENTS.END)`?

Reply:
It's used in a couple of examples here: http://js-sdk-docs.salemove.com/class/Engagement.html

Reply:
engagement.addEventListener(engagement.engagement.EVENTS.END, function() {
      clearInterval(idleInterval);
});

throws the following error after the engagement is ended:

Uncaught (in promise) Error {cause: "internal error"}

Reply:
Is the double `engagement` in your example (`engagement.engagement.EVENTS.END`) a typo that's only here in the comment or also in the code you executed?

Reply:
Seems that not only in the comment. Thanks!